cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/IOUContract.kt

```diff
@@ -46,12 +47,17 @@ class IOUContract : Contract {
                 "The IOU's value must be non-negative." using (out.value > 0)
             }
             is Commands.Destroy -> requireThat {
-                "There should be no IOUState output." using (tx.outputsOfType<IOUState>().isEmpty())
+                "There should be two inputs in general." using (tx.inputs.size == 2)
                 "There should be only one IOUState input." using (tx.inputsOfType<IOUState>().size == 1)
-                val inputIOU = tx.inputsOfType<IOUState>().single()
+                "There should be only one CashState input." using (tx.inputsOfType<CashState>().size == 1)
```

Comment:
I don't think that you need the IOUContract to care about payment. It is up to the involved parties to accept the presence or the form of payment.
Before, you prevented the presence of a payment. I said not to prevent. Now you are imposing the payment, and I say, don't prevent, don't impose.

Reply:
Like Xavier, said; each contract is better off caring about the state it represents, because remember, your IOU contracts module (i.e. state and its contract) can be distributed to other entities (i.e. nodes) and their developers might create an IOU-create flow which doesn't require a payment (for some reason); your current contract implementation prevents that. So let's decouple states from each other and only enforce their relationship in the flow when the transaction is atomic and it requires the existence of both states as inputs.